pkg/dotgit/registry/push/push.go

```diff
@@ -0,0 +1,100 @@
+package push
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"slices"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+	"oras.land/oras-go/v2/registry/remote/retry"
+)
+
+type options struct {
+	force bool
+}
+
+type Option interface {
+	apply(*options)
+}
+
+type forceOption bool
+
+func (f forceOption) apply(opts *options) {
+	opts.force = bool(f)
+}
+
+func WithForce(force bool) Option {
+	return forceOption(force)
+}
+
+func Push(repository, tag, dotgit, token string, opts ...Option) error {
+	options := &options{
+		force: false,
+	}
+
+	for _, opt := range opts {
+		opt.apply(options)
+	}
+
+	ctx := context.TODO()
+
+	repo, err := remote.NewRepository(repository)
+	if err != nil {
+		return errors.Wrapf(err, "create client for %s", repository)
+	}
+	if repo.Reference.Reference != "" {
+		return errors.Errorf("unexpected repository format. expected: %q, actual: %q", []string{"<repository>"}, repository)
+	}
+
+	repo.Client = &auth.Client{
+		Client: retry.DefaultClient,
+		Cache:  auth.NewCache(),
+		Credential: auth.StaticCredential(repo.Reference.Host(), auth.Credential{
+			Username: "user", // Any string but empty
+			Password: token,
+		}),
+	}
+
+	if !options.force {
+		if err := repo.Tags(ctx, "", func(tags []string) error {
```

Comment:
We want to check single tag existence, so does https://pkg.go.dev/oras.land/oras-go/v2/registry/remote#Repository.Exists suffice?

Reply:
They are essentially the same, but Resolve seems sufficient.
https://pkg.go.dev/oras.land/oras-go/v2/registry/remote#Repository.Resolve

https://github.com/MaineK00n/vuls-data-update/compare/2d00d9f2a9335a494df539d3ad6d82d368ee4e00..68e03bd660113ab995c38ece62a7eae8c7ad7041